packages/headless/src/api/commerce/product-listings/v2/product-listing-v2-api-client.ts

```diff
@@ -0,0 +1,65 @@
+import {Logger} from 'pino';
+import {ProductListingV2ThunkExtraArguments} from '../../../../app/product-listing-v2-thunk-extra-arguments';
+import {ProductListingV2AppState} from '../../../../state/commerce-app-state';
+import {PlatformClient} from '../../../platform-client';
+import {PreprocessRequest} from '../../../preprocess-request';
+import {buildAPIResponseFromErrorOrThrow} from '../../../search/search-api-error-response';
+import {
+  buildProductListingV2Request,
+  ProductListingV2Request as ProductListingRequest,
+  ProductListingV2SuccessResponse as ProductListingSuccessResponse,
+} from './product-listing-v2-request';
+
+export interface AsyncThunkProductListingV2Options<
+  T extends Partial<ProductListingV2AppState>
+> {
+  state: T;
+  rejectValue: ProductListingAPIErrorStatusResponse;
+  extra: ProductListingV2ThunkExtraArguments;
+}
+
+export interface ProductListingAPIClientOptions {
```

Comment:
To avoid confusion, I could rename the exports in this file to make sure they include the "V2" part in it.

I wouldn't include "V2" in product listing related public exports, though: this should be transparent to the implementer.

Thoughts?

Reply:
The client has been renamed to `CommerceAPIClient`! We'll most likely do other directory changes in follow-up PRs to remove mentions of `v2` 😄.